protocol/x/clob/e2e/liquidation_deleveraging_test.go

```diff
@@ -20,6 +20,616 @@ import (
+			liquidationConfig: clobtypes.LiquidationsConfig{
+				MaxLiquidationFeePpm: 5_000,
+				FillablePriceConfig:  constants.FillablePriceConfig_Default,
+				PositionBlockLimits:  constants.PositionBlockLimits_Default,
+				SubaccountBlockLimits: clobtypes.SubaccountBlockLimits{
+					// Subaccount may only liquidate $5,000 per block.
+					MaxNotionalLiquidated:    5_000_000_000,
+					MaxQuantumsInsuranceLost: 100_000_000_000_000,
+				},
+			},
+
+			liquidityTiers: constants.LiquidityTiers,
+			perpetuals: []perptypes.Perpetual{
+				constants.BtcUsd_20PercentInitial_10PercentMaintenance,
+			},
+			clobPairs: []clobtypes.ClobPair{constants.ClobPair_Btc},
+
+			expectedSubaccounts: []satypes.Subaccount{
+				{
+					Id: &constants.Carl_Num0,
+					AssetPositions: []*satypes.AssetPosition{
+						{
+							AssetId:  0,
+							Quantums: dtypes.NewInt(50_499_000_000 - 4_950_000_000 - 24_750_000),
+						},
+					},
+					PerpetualPositions: []*satypes.PerpetualPosition{
+						{
+							PerpetualId:  0,
+							Quantums:     dtypes.NewInt(-90_000_000), // -0.9 BTC
+							FundingIndex: dtypes.NewInt(0),
+						},
+					},
+				},
+				{
+					Id: &constants.Dave_Num0,
+					AssetPositions: []*satypes.AssetPosition{
+						{
+							AssetId:  0,
+							Quantums: dtypes.NewInt(54_950_000_000), // $54,950
+						},
+					},
+					PerpetualPositions: []*satypes.PerpetualPosition{
+						{
+							PerpetualId:  0,
+							Quantums:     dtypes.NewInt(90_000_000), // 0.9 BTC
+							FundingIndex: dtypes.NewInt(0),
+						},
+					},
+				},
+			},
+		},
+		`Liquidating long respects subaccount block limit - MaxNotionalLiquidated`: {
+			subaccounts: []satypes.Subaccount{
+				constants.Carl_Num0_1BTC_Short_100000USD,
+				constants.Dave_Num0_1BTC_Long_49501USD_Short,
+			},
+
+			placedMatchableOrders: []clobtypes.MatchableOrder{
+				&constants.Order_Carl_Num0_Id2_Clob0_Buy1BTC_Price50500_GTB10, // Order at $50,500
+			},
+			liquidatableSubaccountIds: []satypes.SubaccountId{constants.Dave_Num0},
+			liquidationConfig: clobtypes.LiquidationsConfig{
+				MaxLiquidationFeePpm: 5_000,
+				FillablePriceConfig:  constants.FillablePriceConfig_Default,
+				PositionBlockLimits:  constants.PositionBlockLimits_Default,
+				SubaccountBlockLimits: clobtypes.SubaccountBlockLimits{
+					// Subaccount may only liquidate $5,000 per block.
+					MaxNotionalLiquidated:    5_000_000_000,
+					MaxQuantumsInsuranceLost: 100_000_000_000_000,
+				},
+			},
+
+			liquidityTiers: constants.LiquidityTiers,
+			perpetuals: []perptypes.Perpetual{
+				constants.BtcUsd_20PercentInitial_10PercentMaintenance,
+			},
+			clobPairs: []clobtypes.ClobPair{constants.ClobPair_Btc},
+
+			expectedSubaccounts: []satypes.Subaccount{
+				{
+					Id: &constants.Carl_Num0,
+					AssetPositions: []*satypes.AssetPosition{
+						{
+							AssetId:  0,
+							Quantums: dtypes.NewInt(100_000_000_000 - 5_050_000_000),
```

Comment:
Here I am seeing more than $5k leaving the asset position. To be clear, this IS expected because we chose to liquidate a number of quantums which corresponds to 5k in notional value? However, the actual price that we matched at (because fillable price can be worse than oracle price) resulted in the subaccount paying more than 5k

Reply:
Was the price of the maker order selected to demonstrate this property? Would probably be more clear if the maker order was at the oracle price

Reply:
yeah you are exactly right. we don't know what the order will get matched at when building the liquidation order, so we use the notional value (oracle price). for consistency, during matching, we also use the notional value (not the actual notional liquidated). otherwise, this inconsistency can lead to issues in DeliverTx.

Reply:
and yeah the maker order in this test was specifically chosen to demonstrate this. let me add a comment